src/planner.rs

```diff
@@ -3,25 +3,33 @@ use std::collections::{HashMap, HashSet, VecDeque};
 use std::fs::{self};
 use std::os::unix::fs::MetadataExt;
 use std::path::PathBuf;
+use std::cmp;
+
 
 pub struct ConexPlanner {
     pub layer_to_files: Vec<(String, Vec<ConexFile>)>,
+    pub split_threshold: usize,
 }
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Default)]
 pub struct ConexFile {
     pub path: PathBuf,
     pub relative_path: PathBuf,
     pub size: usize,
     pub inode: u64,
     pub hard_link_to: Option<PathBuf>,
     pub ctime_nsec: i64,
+    pub start_offset: Option<usize>,
+    pub chunk_size: Option<usize>,
+    pub segment_idx: Option<usize>
 }
 
 impl ConexPlanner {
     pub fn default() -> Self {
         Self {
             layer_to_files: Vec::new(),
+            //30 mb threshold
+            split_threshold: 30 * 1024 * 1024,
```

Comment:
can this be passed in through CLI. this means it needs to be part of `default()` arguments, and then passed in through the Clap App in `main.rs`.